tests/commands/validators_test.py

```diff
@@ -0,0 +1,131 @@
+# encoding=utf-8
+# Copyright © 2017 Dylan Baker
+
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+"""Tests for alot.commands.validators module."""
+
+from __future__ import absolute_import
+
+import os
+import contextlib
+import shutil
+import subprocess
+import tempfile
+import unittest
+
+from alot.commands import validators
+from alot.commands.actions import ValidationFailed
+
+
+@contextlib.contextmanager
+def temporary_directory(suffix='', prefix='', dir=None):
+    """Python3 interface implementation.
+
+    Python3 provides a class that can be used as a context manager, which
+    creates a temporary directory and removes it when the context manager
+    exits. This function emulates enough of the interface of
+    TemporaryDirectory, for this module to use, and is designed as a drop in
+    replacement that can be replaced after the python3 port.
+
+    The only user visible difference is that this does not implement the
+    cleanup method that TemporaryDirectory does.
+    """
+    directory = tempfile.mkdtemp(suffix=suffix, prefix=prefix, dir=dir)
+    yield directory
+    shutil.rmtree(directory)
+
+
+class TestRequireFile(unittest.TestCase):
+    """Tests for the require_file validator."""
+
+    def test_doesnt_exist(self):
+        with temporary_directory() as d:
+            with self.assertRaises(ValidationFailed):
+                validators.require_file(os.path.join(d, 'doesnt-exist'))
+
+    def test_dir(self):
+        with temporary_directory() as d:
+            with self.assertRaises(ValidationFailed):
+                validators.require_file(d)
+
+    def test_file(self):
+        with tempfile.NamedTemporaryFile() as f:
+            validators.require_file(f.name)
+
+    def test_char_special(self):
+        with self.assertRaises(ValidationFailed):
+            validators.require_file('/dev/null')
+
+    def test_fifo(self):
+        with temporary_directory() as d:
+            path = os.path.join(d, 'fifo')
+            subprocess.call(['mkfifo', path])
```

Comment:
Why not `os.mkfifo(path)`? That would allow you to ditch `subprocess` and would probably use the system call directly instead of using the external program.